Update egg-teamspeak3-server.json

xopez opened the pull request:
Allow to users to change filetransfer port, telnet port and ssh port.
I also put the telnet port to the default one and not the voice port.
I saw one problem which I don't know to fix.
For the change of the filestransfer port you need to give an ip. So I put 0.0.0.0
Normally Teamspeak binds `2019-03-22 11:47:57.590315|INFO |VirtualServer |1 |listening on 0.0.0.0:9988, [::]:9988`
But i didn't find a way to give the second one with it. 

Log that you can see it's is starting with custom ports.
```
[Pterodactyl Daemon] Server marked as STARTING
[Pterodactyl Daemon] Checking size of server data directory...
[Pterodactyl Daemon] Disk Usage: 18M / 0M
[Pterodactyl Daemon] Ensuring file permissions.
[Pterodactyl Daemon] Running server preflight.
[Pterodactyl Daemon] Starting server container.
2019-03-22 11:54:14.502446|INFO | | |myTeamSpeak identifier revocation list was downloaded successfully - all related features are activated
2019-03-22 11:54:14.968633|INFO | | |Puzzle precompute time: 521
2019-03-22 11:54:14.968926|INFO |FileManager | |listening on 0.0.0.0:30034
[Pterodactyl Daemon] Server marked as ON
2019-03-22 11:54:14.979194|INFO |VirtualServer |1 |listening on 0.0.0.0:9988, [::]:9988
2019-03-22 11:54:14.979486|INFO |Query | |listening for query on 0.0.0.0:10012, [::]:10012
2019-03-22 11:54:14.979806|INFO |Query | |listening for query ssh on 0.0.0.0:10023, [::]:10023
2019-03-22 11:54:14.979941|INFO |CIDRManager | |updated query_ip_whitelist ips: 127.0.0.1/32, ::1/128,
```

EDIT: I also updated to the latest teamspeak version

EDIT2: I try to change it so it runs with an inifile where you can configure it. So you don't need the startup parameters anymore.

ZeProf2Code commented:
`https:\/\/files.teamspeak-services.com\/releases\/server\/${TS_VERSION}\/teamspeak3-server_linux_amd64-${TS_VERSION}.tar.bz2`
Use this as download URL instead of
`http:\/\/dl.4players.de\/ts\/releases\/${TS_VERSION}\/teamspeak3-server_linux_amd64-${TS_VERSION}.tar.bz2`

xopez commented:
done

ZeProf2Code commented:
👍

xopez commented:
Well, I updated it a little bit. On install creates ts3server.ini with default config. Only creates it, if not there, so users don't need to worry about overwrite when updating the server. Also starts now the binary with the inifile.

xopez commented:
I have finished the new updating way. For short explanation.

- Starts the binary with inifile directly
- `ts3server.ini` and `.ts3server_license_accepted` automatically generated if not existent. Just needed than so that for example the config doesn't get overwritten
- Version info can only get in the `logs/ts3server*_0.log`
- if log not there then install latest
- if there and update on then install newest
- if update off then get the current link and replace the version with the latest from the logfiles
- added more apk requirements
- changed container to `ubuntu:18.04` and script Script Entrypoint Command to `bash` cause I need the command `sort -V/sort --version-sort` which isn't avaiable in ash.

Tested all possible ways for my environment. 
For testing.
If you want to reinstall a old version:
Just install current and start/stop the server. Go to the log `ts3server_*0.log`.
Change the current version to 3.5.0 or something else. Turn update-parameter to `0` and reinstall. Start so can see the old version.
Important. For downgrade delete `ts3server.sqlitedb, ts3server.sqlitedb-shm, ts3server.sqlitedb-wal` cause you can't downgrade a teamspeak db. It gets updated with mostly any version.

ZeProf2Code commented:
what?

xopez commented:
I copied the wrong, changed it already.
My variables are empty. And I don't know why. It works on ubuntu itself quite fine.

parkervcp commented:
@xopez put install logs into a gist or something please. It makes scrolling these difficult.

Instead of make it so a user could decide on what version to install you just defaulted to install the latest or keep an old version. It's a no from me for that alone.

Also never default to accepting a license for a user. It's also no from me for this as well.

parkervcp commented on database/seeds/eggs/voice-servers/egg-teamspeak3-server.json:
```diff
@@ -3,34 +3,34 @@
-            "description": "The version of Teamspeak 3 to use when running the server.",
-            "env_variable": "TS_VERSION",
-            "default_value": "3.5.0",
+            "name": "Update",
```
I commented on defaulting to latest but letting the user choose and you did this instead.

parkervcp commented on database/seeds/eggs/voice-servers/egg-teamspeak3-server.json:
```diff
@@ -3,34 +3,34 @@
-            "script": "#!\/bin\/ash\n# TS3 Installation Script\n#\n# Server Files: \/mnt\/server\napk update\napk add tar curl\n\ncd \/mnt\/server\n\ncurl http:\/\/dl.4players.de\/ts\/releases\/${TS_VERSION}\/teamspeak3-server_linux_amd64-${TS_VERSION}.tar.bz2 | tar xj --strip-components=1",
-            "container": "alpine:3.9",
-            "entrypoint": "ash"
+            "script": "#!\/bin\/bash\r\n# TS3 Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\napt -y update\r\napt -y --no-install-recommends install bzip2 ca-certificates curl grep jq sed tar\r\n\r\ncd \/mnt\/server\r\n\r\n# Check for specific file. Can be there more than one time\r\nfor FILE in logs\/ts3server*_0.log\r\ndo\r\n    # If no log file is there, we can't retrieve the las ...
```
You default to accepting a license for a user. That would make anyone here liable for their user who never had to accept the license.

xopez replied:
Before the license was accepted via command line:
https://github.com/pterodactyl/panel/blob/develop/database/seeds/eggs/voice-servers/egg-teamspeak3-server.json#L11

xopez commented on database/seeds/eggs/voice-servers/egg-teamspeak3-server.json:
```diff
@@ -27,10 +27,37 @@
             "description": "The version of Teamspeak 3 to use when running the server.",
             "env_variable": "TS_VERSION",
-            "default_value": "3.5.0",
+            "default_value": "",
```
Am I understanding you right?
You want that you user can input nothing and latest for latest version or a version to get this specific version?

DaneEveritt commented:
Please re-open this against `pterodactyl:0.7-develop` which has all of the old 0.7.x code, as well as a bunch of TS changes to the egg that probably need to be worked around with these changes anyways.
